v2/federation/fetch_test.go

```diff
@@ -0,0 +1,112 @@
+package federation_test
+
+import (
+	"context"
+	"net"
+	"testing"
+
+	"github.com/spiffe/go-spiffe/v2/bundle/spiffebundle"
+	"github.com/spiffe/go-spiffe/v2/federation"
+	"github.com/spiffe/go-spiffe/v2/internal/test"
+	"github.com/spiffe/go-spiffe/v2/internal/test/fake"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/go-spiffe/v2/svid/x509svid"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFetchBundle_WebPKIAuth(t *testing.T) {
+	td := spiffeid.RequireTrustDomainFromString("domain.test")
+	ca := test.NewCA(t)
+	bundle := spiffebundle.FromX509Bundle(ca.Bundle(td))
+
+	be := fake.NewBundleEndpoint(context.Background(), t, 1025,
+		fake.BEOption.WithTestBundle(bundle))
+	defer be.Shutdown()
+
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
What is the purpose of the cancel-able contexts passed into `federation.FetchBundle` in each test? Is `context.Background()` insufficient? I don't see where we are leveraging cancellation.